### ISSUE TITLE
Convert Trade and Delivery Mission to new project approach

### ISSUE THREAD

**Issue by bevans2000** [mars-sim/mars-sim#1352]
**Is your feature request related to a problem? Please describe.**
The approach to Missions are changing to follow a more generic/step based approach. An example Test Drive mission has proved the concept but the next step is to replace some existing missions.
This issue focuses on the Trade and Delivery missions as they do not involve any EVA

**Describe the goal(s) you have in mind**
A clear and concise description of what you want to work on or others to work on. 
Replacement of existing Mission classes. Metadata will still remain.

**Describe the solution you desire to see**
A clear and concise description of how you would like to approach or implement the design.

**Screenshots**
Provide pictures and screenshots. Use copy (Ctrl-C) and paste (Ctrl-V) to attach pictures here. Feel free to use screenshots or links from other games and applications to illustrate your points. Use markers to circle and highlight important portions in your pictures. 

**Specifications**
- OS version : [e.g. Windows 10, macOS 10.13, Ubuntu 14.04]
- Java version : [e.g. Oracle JDK 17.0.6, AdoptOpenJDK 17.0.6, openjfx 17]
- Major version and build : [e.g. 3.7.0 build 8814]

**Additional context**
Add any other comments, code blocks or screenshots regarding your feature proposal.

**Comment by mokun** [comment]
> Trade and Delivery missions as they do not involve any EVA

Yesterday I discovered issues when a trading mission arrived at a settlement.

The vehicle crew and/or destination settlement settlers have troubles doing EVA.

**Comment by bevans2000** [comment]
> > Trade and Delivery missions as they do not involve any EVA
> 
> Yesterday I discovered issues when a trading mission arrived at a settlement.
> 
> The vehicle crew and/or destination settlement settlers have troubles doing EVA.
> 

Don't try and fix it because it needs resolving by the bigger problem of who controls visiting Units. We have an issue for that already.

**Comment by mokun** [comment]
> resolving by the bigger problem of who controls visiting Units.

Right. I merely pointed that out here that EVA ops are still needed for unloading and may be loading cargo at the destination settlement. 

Supposedly they are less complicated than having to do it right at the middle of a mission.

I see that we will need not just Weather Station and Communication Tower Construction mission but also Resupply mission for those remote locations. EVA ops will be needed as well.

**Comment by mokun** [comment]
Btw, where is the xml file that define the mission preference for some missions ?

**Comment by bevans2000** [comment]
What preferences do you mean?

**Comment by mokun** [comment]
The preference scores to a mission, science subject, etc. for each settlement.

**Comment by bevans2000** [comment]
Right, I think it is calculated from a combination of the Settlement size and Objectives of the Reporting Authority.

**Comment by mokun** [comment]
I'll look forward to seeing the new Trade and Delivery under the new approach.

After that, I can base on Delivery mission and reuse the code to convert the Emergency Resupply mission using drones for supply drops.
